pkg/workloads/cortex/lib/storage/s3.py

```diff
@@ -27,14 +27,16 @@
 
 
 class S3(object):
-    def __init__(self, bucket=None, region=None, client_config={}):
+    def __init__(self, bucket=None, region=None, client_config={}, anonymous=False):
         self.bucket = bucket
         self.region = region
 
         if client_config is None:
             client_config = {}
 
-        if region is not None:
+        if anonymous:
+            client_config["config"] = botocore.client.Config(signature_version=botocore.UNSIGNED)
+        elif region is not None:
```

Comment:
can you have anonymous=True and region specified at the same time?

Reply:
Yes, apparently, that works okay.